packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -5,104 +5,173 @@
 
-  const StatusIcon = StatusIconMap[type];
-
   return (
     <Box
+      as='div'
       className={cx('iui-alert', className)}
       data-iui-status={type}
       data-iui-variant={isSticky ? 'sticky' : undefined}
       ref={ref}
       {...rest}
     >
-      <StatusIcon className='iui-alert-icon' />
-      <Box as='span' className='iui-alert-message'>
-        {children}
-        {clickableText && (
-          <Box
-            as='a'
-            {...clickableTextProps}
-            className={cx('iui-alert-link', clickableTextProps?.className)}
-          >
-            {clickableText}
-          </Box>
-        )}
-      </Box>
-
-      {onClose && (
-        <Box
-          as='button'
-          onClick={onClose}
-          aria-label='Close'
-          type='button'
-          className='iui-alert-button'
-        >
-          <Box as='span' className='iui-alert-button-icon' aria-hidden>
-            <SvgCloseSmall />
-          </Box>
-        </Box>
-      )}
+      <AlertContext.Provider value={{ type }}>{children}</AlertContext.Provider>
     </Box>
   );
-}) as PolymorphicForwardRefComponent<'div', AlertProps>;
+}) as PolymorphicForwardRefComponent<'div', AlertOwnProps>;
+AlertComponent.displayName = 'Alert';
+
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  const { type } = useSafeContext(AlertContext);
+
+  const StatusIcon = StatusIconMap[type];
+
+  return (
+    <Icon fill={type} ref={ref} {...rest}>
+      {children ?? <StatusIcon />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', React.ComponentProps<typeof Icon>>;
+AlertIcon.displayName = 'Alert.Icon';
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+const AlertMessage = polymorphic.span('iui-alert-message');
+AlertMessage.displayName = 'Alert.Message';
+
+// ----------------------------------------------------------------------------
+// Alert.Action component
+
+const AlertAction = React.forwardRef((props, ref) => {
+  const { children, className, ...rest } = props;
+
+  return (
+    <Box
+      as={(!!props.href ? 'a' : 'button') as 'a'}
+      className={cx('iui-alert-link', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'a'>;
+AlertAction.displayName = 'Alert.Action';
+
+// ----------------------------------------------------------------------------
+// Alert.CloseButton component
+
+const AlertCloseButton = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  return (
+    <IconButton
+      styleType='borderless'
+      size='small'
+      aria-label='Close'
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgCloseSmall />}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<'button'>;
```

Comment:
This is more a question, but since we are returning a specific component here `IconButton`, the Polymorphic element default `button` doesn't really do anything right?

Reply:
it does. It allows passing all native `button` attributes such as `onClick`, `type`, `onFocus`. Basically it should match the default element that `IconButton` uses.